### DISCUSSION TITLE
How do I handle multiple slashes in the URL path leading, mid-path or trailing, similar to non strict slashes in Flask.

### DISCUSSION THREAD

**Discussion by mehmetg** [fastapi/fastapi#7312] [Questions] [answered]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from fastapi import FastAPI

app = FastAPI()


@app.get("/hello/x")
def read_root():
    return {"Hello": "World"}
```

### Description

```bash
curl http://localhost:8000/hello/x --> is OK
curl http://localhost:8000///hello///x --> throws 404
```

**Comment by mehmetg** [comment]
I handled it using the following redirect middleware. Is there any downside to this? 
 
```python
import re

from starlette.datastructures import URL
from starlette.responses import RedirectResponse
from starlette.types import ASGIApp, Receive, Scope, Send

repeated_quotes = re.compile(r'//+')


class HttpUrlRedirectMiddleware:
  """
  This http middleware redirects urls with repeated slashes to the cleaned up
  versions of the urls
  """

  def __init__(self, app: ASGIApp) -> None:
    self.app = app

  async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:

    if scope["type"] == "http" and repeated_quotes.search(URL(scope=scope).path):
      url = URL(scope=scope)
      url = url.replace(path=repeated_quotes.sub('/', url.path))
      response = RedirectResponse(url, status_code=307)
      await response(scope, receive, send)
    else:
      await self.app(scope, receive, send)
```

**Comment by NightMachinery** [comment]
[**@mehmetg**](https://github.com/mehmetg) commented on [Sep 25, 2020, 11:36 PM GMT+3:30](https://github.com/tiangolo/fastapi/issues/2090#issuecomment-699129618 "2020-09-25T20:06:08Z - Replied by Github Reply Comments"):
> I handled it using the following redirect middleware. Is there any downside to this?
> 
> ```python
> import re
> 
> from starlette.datastructures import URL
> from starlette.responses import RedirectResponse
> from starlette.types import ASGIApp, Receive, Scope, Send
> 
> repeated_quotes = re.compile(r'//+')
> 
> 
> class HttpUrlRedirectMiddleware:
>   """
>   This http middleware redirects urls with repeated slashes to the cleaned up
>   versions of the urls
>   """
> 
>   def __init__(self, app: ASGIApp) -> None:
>     self.app = app
> 
>   async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
> 
>     if scope["type"] == "http" and repeated_quotes.search(URL(scope=scope).path):
>       url = URL(scope=scope)
>       url = url.replace(path=repeated_quotes.sub('/', url.path))
>       response = RedirectResponse(url, status_code=307)
>       await response(scope, receive, send)
>     else:
>       await self.app(scope, receive, send)
> ```

How do you add this middleware to fastapi?

**Comment by Kludex** [comment, marked as answer]
@NightMachinary 

```python
from fastapi import FastAPI

from someplace import HttpUrlRedirectMiddleware

app = FastAPI()
app.add_middleware(HttpUrlRedirectMiddleware)
```

**Reply by cyysky** [reply]
```
import re
from starlette.middleware.base import BaseHTTPMiddleware

class NormalizeSlashMiddleware(BaseHTTPMiddleware):
    async def dispatch(self, request, call_next):
        path = request.scope["path"]
        normalized_path = re.sub(r'/{2,}', '/', path)
        request.scope["path"] = normalized_path
        return await call_next(request)

app.add_middleware(NormalizeSlashMiddleware)
```

Try this ;-)

**Comment by mehmetg** [comment]
I assume this is an acceptable way to handle this as there were no objections. I have been using it in production and have not received any complaints.
Closing.

**Comment by tiangolo** [comment]
Thanks for the help and discussion here everyone!

And thanks for coming back to close the issue @mehmetg ☕ 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
